src/emqx_shared_sub.erl

```diff
@@ -228,16 +230,17 @@ is_ack_required(Msg) -> ?NO_ACK =/= get_group_ack(Msg).
 -spec(maybe_nack_dropped(emqx_types:message()) -> boolean()).
 maybe_nack_dropped(Msg) ->
     case get_group_ack(Msg) of
-        ?NO_ACK -> false;
-        {_Group, Sender, Ref} -> ok == nack(Sender, Ref, dropped)
+        ?NO_ACK                       -> false;
+        {fresh, _Group, Sender, Ref}   -> ok == nack(Sender, Ref, dropped);
```

Comment:
Once we changed the data structure, we must add some compatibility codes for hot-upgrade.

Reply:
How do I do this?

Reply:
I think we might need to revert this PR as well, because it also adds a new field into structure

https://github.com/emqx/emqx/pull/7807

Reply:
I've added a fix PR (#8028) for #7807.

And I've added a commit for this PR. This change is compatible, because this structure is sent as a message header, and receiver (emqx_session) doesn't use this header in any way. It's opaque for the receiver

The only issue here is that sender (emqx_shared_sub) can send an old version of structure and then get updated before he receive a reply. This occurs only during update, because restarting solves it, and lost ACK's can just lost a message in a very rare situation, thus this is tolerable